packages/frontend/src/__tests__/textEditor.js

```diff
@@ -1,22 +1,36 @@
-import { screen, render } from '@testing-library/react'
+import { screen, render, waitFor } from '@testing-library/react'
+import { useState } from 'react'
 import userEvent from '@testing-library/user-event'
 import TextEditor from '../components/textEditor'
 
 const loremText =
     'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum'
 
-const renderTextEditor = (content, setContent, autoFocus) => {
-    return render(
+const TestWrapper = () => {
+    const [content, setContent] = useState('')
+    return (
         <TextEditor
             content={content}
             setContent={setContent}
-            autoFocus={autoFocus}
+            autoFocus={true}
         />
     )
 }
+
+const renderTextEditor = () => {
+    return render(<TestWrapper />)
+}
+
 test('type text into textbox', async () => {
-    renderTextEditor(loremText, jest.fn(), true)
-    expect(screen.getByText(loremText)).toBeInTheDocument()
+    renderTextEditor()
+    const textbox = screen.getByRole('textbox')
+    userEvent.type(textbox, loremText)
+    await waitFor(
```

Comment:
instead of a `waitFor` clause you can use a promise like this:
```js
userEvent.type(textbox, loremText)
  .then(() => expect(textbox).toHaveValue(loremText))
```

Reply:
thanks, much cleaner this way